### PR TITLE
Move where PreviousMarks is rendering

### LINKED ISSUE #2003: Previous marks block drawing in subsequent workflow steps
Issue is in first comment.

### PR DESCRIPTION
**Pull request by srallen**
_Please request review from `@zooniverse/frontend` team. If PR is related to design, please request review from `@beckyrother` in addition._ 

Package: lib-classifier

Closes #2003 

This moves where `PreviousMarks` is rendering down to `InteractionLayer` component and is correctly rendered in order but does not block subsequent drawing tasks from being able to create new marks. 

Testable with:
- Transcription task: https://localhost:8080/?project=11300&env=production
- Multiple drawing tasks: https://localhost:8080/?project=7170&env=production

You may notice that it's a bit redundant to pass along the task along with its properties to `InteractionLayer` and I would agree, however, I spent a while refactoring today only to be blocked by tests that were difficult to get passing again because of how data is being mocked for them. I'm going to scope this to only be the bug fix and subsequent refactoring to be separate. I have a new idea on how to go about refactoring that I think will really clear up this code, but I'm not sure when I'll be able to next have time for it. I'd like this review in scope of the bug fix only so we can move ahead with beta testing.

# Review Checklist

## General

- [ ] Are the tests passing locally and on Travis?
- [ ] Is the documentation up to date?

## Components
- [ ] Has a storybook story been created or updated?
- [ ] Is the component accessible? 
  - [ ] Can it be used with a screen reader? [BBC guide to testing with VoiceOver](https://bbc.github.io/accessibility-news-and-you/accessibility-and-testing-with-voiceover-os.html)
  - [ ] Can it be used from the keyboard? [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - [ ] Is it passing accessibility checks in the storybook?

## Apps

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `yarn panic && yarn bootstrap` or `docker-compose up --build` and app works as expected?

## Publishing

- [ ] Is the changelog updated?
- [ ] Are the dependencies updated for apps and libraries that are using the newly published library?

## Post-merging

- [ ] Did the app deploy to https://frontend.preview.zooniverse.org/projects/:project-name/:owner or https://frontend.preview.zooniverse.org/about?
- [ ] Is the new feature working or bug now fixed?
  - [ ] Is there a Talk or blog post written to announce the new feature(s)?
- [ ] Is the design working across browsers (Firefox, Chrome, Edge, Safari) and mobile?
  - [ ] Is this approved by our designer?
- [ ] Is this ready for production deployment?

### CODE REVIEW

**Comment by srallen** [comment]
@eatyourgreens let me know if you can review this tomorrow if possible. Thanks!

**Comment by eatyourgreens** [comment]
Sure, I'll take a look. The changes here might reverse our decision to keep active drawing marks and previous marks separate: https://github.com/zooniverse/front-end-monorepo/blob/master/docs/arch/adr-22.md#the-subject-viewer

I think styling `PreviousMarks` with `pointer-events: none` is a simpler solution, but maybe I've missed some side effects of that change? See #2006.

**Review by eatyourgreens — CHANGES_REQUESTED** [review]
On Worlds of Wonder (https://localhost:8080/?project=7170&env=production) I can place a rectangle, but then can't add points on top of it for the second task. `PreviousMarks` is rendered on top of the `DrawingCanvas` component, blocking it from receiving pointer events.

File: packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/InteractionLayer/InteractionLayer.js [hunk 1]
```diff
@@ -128,23 +130,25 @@ function InteractionLayer ({
...
+          onPointerUp={onPointerUp}
+          touch-action='none'
+        />}
       <TranscribedLines
         scale={scale}
       />
       <SubTaskPopup
         onDelete={inactivateMark}
       />
+      <PreviousMarks scale={scale} />
```
Comment (eatyourgreens) [inline packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/InteractionLayer/InteractionLayer.js:151]: `PreviousMarks` is trapping pointer events here, preventing the earlier `DrawingCanvas` from receiving them.

**Comment by srallen** [comment]
> Sure, I'll take a look. The changes here might reverse our decision to keep active drawing marks and previous marks separate: https://github.com/zooniverse/front-end-monorepo/blob/master/docs/arch/adr-22.md#the-subject-viewer

It doesn't, but I guess doesn't clearly appear that way because the refactoring that I have in mind isn't complete. What I have in mind would make the separation of concerns very clear. 

> I think styling PreviousMarks with pointer-events: none is a simpler solution, but maybe I've missed some side effects of that change? See #2006.

Technically this does fix it, but it makes it fuzzy as to what controls what and why this has to be styled this way. I'll accept it since the beta needs to move forward, but long term I'd like to still see a refactoring of how this all is rendering.

**Pull request closed without merging**